I²S Media Player won't build, I2S_DAC_CHANNEL_BOTH_EN undeclared

### The problem

The change to I²S Audio Component in 2023.4.2 appears to have broken the ability to build even after the YAML has been updated due to I2S_DAC_CHANNEL_BOTH_EN being undeclared.

### Which version of ESPHome has the issue?

2023.4.2

### What type of installation are you using?

Home Assistant Add-on

### Which version of Home Assistant has the issue?

2023.4.6 / Supervisor 2023.04.1 / OS 10.1

### What platform are you using?

ESP32S2

### Board

Wemos S2 mini

### Component causing the issue

i2s_audio

### Example YAML snippet

```yaml
esp32:
  board: esp32-s2-saola-1
  framework:
    type: arduino

i2s_audio:
  i2s_lrclk_pin: GPIO37
  i2s_bclk_pin: GPIO35

media_player:
  - platform: i2s_audio
    name: A040
    dac_type: external
    i2s_dout_pin: GPIO33
    mode: mono
```


### Anything in the logs that might be useful for us?

```txt
Build output below...

INFO Reading configuration /config/esphome/audio040.yaml...
INFO Generating C++ source...
INFO Compiling app...
Processing audio040 (board: esp32-s2-saola-1; framework: arduino; platform: platformio/espressif32 @ 5.3.0)
--------------------------------------------------------------------------------
HARDWARE: ESP32S2 240MHz, 320KB RAM, 4MB Flash
 - toolchain-riscv32-esp @ 8.4.0+2021r2-patch5 
 - toolchain-xtensa-esp32s2 @ 8.4.0+2021r2-patch5
Dependency Graph
|-- AsyncTCP-esphome @ 1.2.2
|-- WiFi @ 2.0.0
|-- FS @ 2.0.0
|-- Update @ 2.0.0
|-- ESPAsyncWebServer-esphome @ 2.1.0
|   |-- AsyncTCP-esphome @ 1.2.2
|-- DNSServer @ 2.0.0
|-- ESPmDNS @ 2.0.0
|-- noise-c @ 0.1.4
|   |-- libsodium @ 1.10018.1
|-- ArduinoJson @ 6.18.5
|-- WiFiClientSecure @ 2.0.0
|-- HTTPClient @ 2.0.0
|-- ESP32-audioI2S @ 2.0.6
Compiling /data/audio040/.pioenvs/audio040/src/esphome/components/i2s_audio/media_player/i2s_audio_media_player.cpp.o
Compiling /data/audio040/.pioenvs/audio040/src/esphome/components/socket/bsd_sockets_impl.cpp.o
src/esphome/components/i2s_audio/media_player/i2s_audio_media_player.cpp: In member function 'void esphome::i2s_audio::I2SAudioMediaPlayer::start_()':
src/esphome/components/i2s_audio/media_player/i2s_audio_media_player.cpp:144:46: error: 'I2S_DAC_CHANNEL_BOTH_EN' was not declared in this scope
     this->audio_ = make_unique<Audio>(false, I2S_DAC_CHANNEL_BOTH_EN, this->parent_->get_port());
                                              ^~~~~~~~~~~~~~~~~~~~~~~
Compiling /data/audio040/.pioenvs/audio040/src/esphome/components/socket/lwip_raw_tcp_impl.cpp.o
src/esphome/components/i2s_audio/media_player/i2s_audio_media_player.cpp:144:46: note: suggested alternative: 'I2S_CHANNEL_STEREO'
     this->audio_ = make_unique<Audio>(false, I2S_DAC_CHANNEL_BOTH_EN, this->parent_->get_port());
                                              ^~~~~~~~~~~~~~~~~~~~~~~
                                              I2S_CHANNEL_STEREO
*** [/data/audio040/.pioenvs/audio040/src/esphome/components/i2s_audio/media_player/i2s_audio_media_player.cpp.o] Error 1
========================== [FAILED] Took 5.84 seconds ==========================
```


### Additional information

_No response_

--- a/esphome/components/i2s_audio/media_player/i2s_audio_media_player.cpp
+++ b/esphome/components/i2s_audio/media_player/i2s_audio_media_player.cpp
@@ -141,7 +141,7 @@ void I2SAudioMediaPlayer::start_() {
     this->audio_ = make_unique<Audio>(true, this->internal_dac_mode_, this->parent_->get_port());
   } else {
 #endif
-    this->audio_ = make_unique<Audio>(false, I2S_DAC_CHANNEL_BOTH_EN, this->parent_->get_port());
+    this->audio_ = make_unique<Audio>(false, 3, this->parent_->get_port());
 
     i2s_pin_config_t pin_config = this->parent_->get_pin_config();
     pin_config.data_out_num = this->dout_pin_;

